woocommerce/woocommerce
# feat: add product review autocomplete fields

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/51906 .

I considered refactoring the whole form to use the `woocommerce_form_field()` function, instead of the current approach.
I thought that the `woocommerce_form_field()` function could be a better substitute for the current functionality.
However, that would be a breaking change for themes & styles, so I opted for not doing that.



### How to test the changes in this Pull Request:



- Ensure you have a simple product on your installation (e.g.: you can use the "Album" product from the product templates after installing WooCommerce)
- Go to wp-admin > WooCommerce > Settings > Products > General
- Check the box next to Enable product reviews
<img width="627" alt="Screenshot 2024-10-08 at 5 16 09 PM" src="https://github.com/user-attachments/assets/3aaa5d44-da11-4945-9093-397bebae7a9b">

- Click the "Save changes" button to save the new settings
- Log out
- Go to the product page for the simple product you created earlier
- Click on the "Reviews" tab on the product page
- Inspect the page with your browser
- The "Name" and "Email" inputs should now include an `autocomplete` attribute of `name` and `email`, respectively
- Your browser should suggest autocomplete details

Before:
<img width="662" alt="Screenshot 2024-10-08 at 5 15 17 PM" src="https://github.com/user-attachments/assets/6685258f-f5b9-44c2-94f5-0487e0498071">

After:
<img width="801" alt="Screenshot 2024-10-08 at 5 14 05 PM" src="https://github.com/user-attachments/assets/d08ce0b0-361e-45c1-a8e7-d64414aecc1c">



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
